src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutorExtensions.cs

```diff
@@ -31,7 +33,16 @@ public static async Task<IDelayedException> TryExecuteAsync(this IFunctionExecut
                     logger = loggerFactory.CreateLogger(LogCategories.CreateFunctionCategory(functionInstance.FunctionDescriptor.LogName));
                 }
 
+                if (functionInstance is FunctionInstance instance)
+                {
+                    retryContext.RetryCount = attempt;
+                    retryContext.Instance = instance;
```

Comment:
nit: move this below condition ```if (functionInstance.FunctionDescriptor.RetryStrategy == null)``` and move ```retryContext.Exception = functionResult?.Exception;``` inside if block  , this will avoid initializing retry context if RetryStrategy is not specified.

Reply:
it does not work as we need to set the properties before the execution.

Reply:
Pragna does have a point though - we should only set this if the function actually uses a RetryStrategy, so we should expand the if clause here to also check RetryStrategy != null. Furthermore, can we also only create/set the context if attempt > 0? I.e. ExecutionContext.RetryContext is only set for failed invocations when we're doing retries? That allows us to avoid overhead of object creations for normal invocations.

We'd also move the creation of the retry context into this block as well then: retryContext = retryContext ?? new RetryContext();

Reply:
Yes, I agree. RetryContext should only be initialized if "RetryStrategy" is used by a function